webpack/scenes/ContentViews/Create/CreateContentViewForm.js

```diff
@@ -0,0 +1,154 @@
+import { STATUS } from 'foremanReact/constants';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import { Form, FormGroup, TextInput, TextArea, Checkbox, Radio, ActionGroup, Button } from '@patternfly/react-core';
+import { createContentView } from '../ContentViewsActions';
+import { selectCreateContentViews, selectCreateContentViewStatus, selectCreateContentViewError } from './ContentViewCreateSelectors';
+import { labelComponent, labelComposite, labelDependencies, labelAutoPublish } from './CreateContentViewFormComponents';
+
+const CreateContentViewForm = ({ setModalOpen }) => {
+  const dispatch = useDispatch();
+  const [name, setName] = useState('');
+  const [label, setLabel] = useState('');
+  const [description, setDescription] = useState('');
+  const [composite, setComposite] = useState(false);
+  const [component, setComponent] = useState(true);
+  const [autoPublish, setAutoPublish] = useState(false);
+  const [importOnly, setImportOnly] = useState(false);
+  const [dependencies, setDependencies] = useState(false);
+  const [redirect, setRedirect] = useState(false);
+  const [redirectUrl, setRedirectURL] = useState('/labs/content_views');
+
+  const response = useSelector(selectCreateContentViews);
+  const status = useSelector(selectCreateContentViewStatus);
+  const error = useSelector(selectCreateContentViewError);
+
+  useEffect(() => {
+    const { id } = response;
+    if (id && status === STATUS.RESOLVED) {
+      setRedirect(true);
+      setRedirectURL(`/labs/content_views/${id}`);
+    }
+  }, [JSON.stringify(response), status, error]);
+
+  const onSave = () => dispatch(createContentView({
+    name,
+    label,
+    description,
+    composite,
+    dependencies,
+    auto_publish: (autoPublish && composite),
+    import_only: importOnly,
+  }));
+
+  useEffect(
+    () => {
+      setLabel(name.replace(/ /g, '_'));
+    },
+    [name],
+  );
+
+  if (redirect) {
+    return (<Redirect to={redirectUrl} />);
```

Comment:
Something about setting the `redirectUrl` as a hook still doesn't feel right, what about doing this?
```js
  if (redirect) {
    const { id } = response;
    return (<Redirect to={`/labs/content_views/${id}`} />);
```
and remove the hook from the effect above